routing/route/route.go

```diff
@@ -311,6 +311,11 @@ func (r *Route) TotalFees() lnwire.MilliSatoshi {
 	return r.TotalAmount - r.Hops[len(r.Hops)-1].AmtToForward
 }
 
+// Amt is the amount received by the final hop of this route.
+func (r *Route) Amt() lnwire.MilliSatoshi {
+	return r.TotalAmount - r.TotalFees()
```

Comment:
I think this is easier to read, and fewer lines of code.

Reply:
I think we should have panics when the route is empty. It is an undefined state that goes unnoticed.

I don't like `r.TotalAmount - (r.TotalAmount - amtToForward)`. Unnecessary calculation.

You could also make `TotalFees` use `ReceiverAmt`, where `ReceiverAmt` returns ` r.Hops[len(r.Hops)-1].AmtToForward`.

Reply:
I also think panicing is okay, since we should catch zero hop routes. Went with it: https://github.com/lightningnetwork/lnd/pull/3970/commits/8856582cce34257396b0ce7bf336360e9d3e701e

Reply:
The solution leads to a panic during integration tests, since we use zero hop routes for to expose on the RPC if we don't have the route stored: https://github.com/lightningnetwork/lnd/blob/master/rpcserver.go#L5184

Maybe just leave the zero hop check in place, and worry about this later?

Reply:
If we don't widely rely on zero-hop routes, isn't it just as easy to fix the code above? Make `route` a pointer and only set fee etc if not nil?

Reply:
For some use case a no-hop route can still be valid, it feels like. And since this is the kind of thing that might open another can of worms, I don't feel like doing it in this PR. wdyt?

Reply:
I don't think there is a use case for which a no-hop route is valid. But ofc, this can be done in a separate pr. It is pre-existing